fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PullUpPredicateRule.java

```diff
@@ -0,0 +1,227 @@
+
+    @Override
+    public OptExpression rewrite(OptExpression root, TaskContext taskContext) {
+        RewriteContext rewriteContext = new RewriteContext();
+        root.getOp().accept(new Visitor(), root, rewriteContext);
+
+        //Convert all remaining predicates to LogicalFilter
+        convertPredicateToLogicalFilter(root, rewriteContext);
+        return root;
+    }
+
+    private static class Visitor extends OptExpressionVisitor<Void, RewriteContext> {
+
+        public Visitor() {
+        }
+
+        @Override
+        public Void visit(OptExpression optExpression, RewriteContext context) {
+            for (int childIdx = 0; childIdx < optExpression.getInputs().size(); ++childIdx) {
+                OptExpression child = optExpression.inputAt(childIdx);
+
+                RewriteContext rewriteContext = new RewriteContext();
+                child.getOp().accept(this, child, rewriteContext);
+                OptExpression c = handleLegacyPredicate(optExpression.getOutputColumns(), child, rewriteContext);
+                optExpression.setChild(childIdx, c);
+
+                context.columnRefToConstant.putAll(rewriteContext.columnRefToConstant);
+            }
+
+            return null;
+        }
+
+        public OptExpression handleLegacyPredicate(ColumnRefSet parentInputColumns, OptExpression optExpression,
+                                                   RewriteContext context) {
+            OptExpression root = optExpression;
+            if (parentInputColumns != null) {
+                Iterator<Map.Entry<ColumnRefOperator, ScalarOperator>> iterator =
+                        context.columnRefToConstant.entrySet().iterator();
+                while (iterator.hasNext()) {
+                    Map.Entry<ColumnRefOperator, ScalarOperator> entry = iterator.next();
+                    if (!parentInputColumns.contains(entry.getKey())) {
+
+                        // If the output columns does not contain this predicate,
+                        // leave the predicate under this node to provide more possibilities for subsequent predicate push down
+                        root = OptExpression.create(new LogicalFilterOperator(
+                                BinaryPredicateOperator.eq(entry.getKey(), entry.getValue())), root);
+                        iterator.remove();
+                    }
+                }
+            }
+            return root;
+        }
+
+        @Override
+        public Void visitLogicalFilter(OptExpression optExpression, RewriteContext context) {
+            OptExpression child = optExpression.inputAt(0);
+            child.getOp().accept(this, child, context);
+
+            LogicalFilterOperator filterOperator = (LogicalFilterOperator) optExpression.getOp();
+            List<ScalarOperator> inputPredicates = Utils.extractConjuncts(filterOperator.getPredicate());
+            for (ScalarOperator scalar : inputPredicates) {
+                if (Utils.isConstantEqualPredicate(scalar)) {
```

Comment:
I think we could also support a > 1, a < 1.

Reply:
Yes, it can be done in theory, but it needs to be combined with multiple equivalent predicates. @Seaven also raised the same question "use ScalarEquivalenceExtractor to deriver more predicate", this function is planned to be supported in subsequent PRs, not for This PR adds this complex functionality